scripts/sort-makefile-am-sources.pl

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/perl
+
+while (<>) {
+    # print "read1: $_";
+    if (! /^(nodist_)?tests_test.*_SOURCES/) {
+        print;
+        next;
+    }
+    print;
+    # accumulate files and prep for sorting
+    my %files;
+    while (<>) {
+        # print "read2: $_";
+        chop;
+        /\s*(tests\/stub_|tests\/test)?(\S+)(\s+\\\s*)?$/ || die "no parse";
```

Comment:
The latest sorting I have been trying to work towards just sorts each line in the set into alphanumeric order. The only part ignored is the line-folding characters.  No need to slice the path up before sort comparison.

Reply:
The idea is to try and keep related files close together. So that for instance,

```
ufsdump_SOURCES = \
# [...] \
        tests/stub_fd.cc \
        fd.h \
# [...]
etc.
```
Without this, stubs would be all on their own, reducing understandability

Reply:
The purpose behind listing .h like that one is to ensure they get distributed. Since the unit tests are only listing files that they need to link to most of the .h there are irrelevant and will be removed "any day now". Same goes for extra binaries like ufsdump - the fd.h file is distributed via squid_SOURCES, and need not be listed there.

AIUI the modern auto-tools should be actively determining file dependencies from the compiler output already. So we should no longer need to list .h at all in the Makefile, but do so because none of us has verified that dependency detection reliability.

Reply:
I'll check it in a different PR